alica_engine/include/engine/AlicaContext.h

```diff
@@ -417,6 +425,11 @@ class AlicaContext
      * @note Is called when setOption or setOptions is successfully called.
      */
     void reloadConfig();
+
+    /*
+     * Get communication Handlers
+     */
+    std::shared_ptr<AlicaCommunicationHandlers> getCommunicationHandlers();
```

Comment:
This don't need to be shared pointer, just return the structure as it is. It only contains function pointers and don't need allocation on heap.